### PR TITLE
CORE-668: Add asBlockChainDBModelCurrency in Java Core Crypto

### PR DESCRIPTION
**Pull request by michael-brd**

### CODE REVIEW

**Comment by EBGToo** [comment]
I pushed a change to the 'code' case for currency and currency denomination.  For Swift.  If I shouldn't ursurp this PR, for Java, with a Swift change, we can unwind.  A change to Java is implied with this Swift change.

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 1]
```diff
@@ -103,10 +103,10 @@ public final class System {
                          (name: "Gwei",  code: "gwei", decimals:  9, symbol: BlockChainDB.Model.lookupSymbol ("gwei")),
                          (name: "Ether", code: "eth",  decimals: 18, symbol: BlockChainDB.Model.lookupSymbol ("eth"))]),
 
-        (id: System.makeCurrencyIdentifierERC20 ("ethereum-mainnet", BlockChainDB.Model.addressBRDMainnet), name: "BRD Token", code: "BRD", type: "erc20", blockchainID: "ethereum-mainnet",
+        (id: System.makeCurrencyIdentifierERC20 ("ethereum-mainnet", BlockChainDB.Model.addressBRDMainnet), name: "BRD Token", code: "brd", type: "erc20", blockchainID: "ethereum-mainnet",
          address: BlockChainDB.Model.addressBRDMainnet, verified: true,
-         demoninations: [(name: "BRD Token INT", code: "BRDI",  decimals:  0, symbol: "brdi"),
-                         (name: "BRD Token",     code: "BRD",   decimals: 18, symbol: "brd")]),
+         demoninations: [(name: "BRD Token INT", code: "brdi",  decimals:  0, symbol: "brdi"),
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: Any issue with this (code formatting) diverging from `currencyToDefaultBaseUnit` and `asBlockChainDBModelCurrency` behaviour?

**Comment by michael-brd** [comment]
> I pushed a change to the 'code' case for currency and currency denomination. For Swift. If I shouldn't ursurp this PR, for Java, with a Swift change, we can unwind. A change to Java is implied with this Swift change.

Made the Java change. Have an open question to you but otherwise good to go.

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 1 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: Looks like it.  This:
```
                let code         = code.uppercased()
```
should be 'lowercased()'?

Comment (michael-brd) [reply Swift/BRCrypto/BRCryptoSystem.swift]: > Looks like it. This:
> 
> ```
>                 let code         = code.uppercased()
> ```
> 
> should be 'lowercased()'?

Can you update the Swift? I'll mirror in Java. You have the "mental model" here as to how these should be.

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: And these:
```
                            (name: "\(code)_INTEGER",
                                code: "\(code)_INTEGER",
                                decimals: 0,
                                symbol: "\(code)I"),   // BRDI -> BaseUnit

```
should have the code and symbol be "\(code)i" ?

**Comment by EBGToo** [comment]
Let me update the Swift.

**Comment by EBGToo** [comment]
> Should code be lowercased here?

Still I missed something?

**Comment by michael-brd** [comment]
> > Should code be lowercased here?
> 
> Still I missed something?

I'm not sure :), that's the question. What case is code supposed to be? `asBlockChainDBModelCurrency` forces it to uppercase.

File: Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Blockchains.java [hunk 2]
```diff
@@ -250,19 +206,41 @@
...
+
+    /* package */
+    static List<CurrencyDenomination> makeCurrencyDemominationsErc20 (String code, UnsignedInteger decimals) {
+        String name = code.toUpperCase(Locale.ROOT);
+        code = code.toLowerCase(Locale.ROOT);
+
+        return ImmutableList.of(
+                new CurrencyDenomination(
+                        String.format(Locale.ROOT, "%s Token INT", name),
+                        String.format(Locale.ROOT, "%si", code),
```
Comment (EBGToo) [inline Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Blockchains.java]: I don't think these are always ERC20 tokens (although in practice I think they are).

File: Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/System.java [hunk 3]
```diff
@@ -226,6 +227,36 @@ static System create(ScheduledExecutorService executor,
...
+                                                                                           String code,
+                                                                                           String type,
+                                                                                           UnsignedInteger decimals) {
+        int index = uids.indexOf(':');
+        if (index == -1) return Optional.absent();
+
+        type = type.toLowerCase(Locale.ROOT);
+        if (!"erc20".equals(type) && !"native".equals(type)) return Optional.absent();
+
+        code = code.toUpperCase(Locale.ROOT);
```
Comment (EBGToo) [inline Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/System.java]: No lower-cased.

File: Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Blockchains.java (hunk 2 above)
Comment (michael-brd) [reply Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Blockchains.java]: I don't follow the comment here; isn't this an exact replica of `makeCurrencyDemominationsERC20` that you just added in Swift?

**Pull request merged**